src/controller/helpers/zclFrameConverter.ts

```diff
@@ -8,10 +8,12 @@ interface KeyValue {[s: string]: number | string}
 // This leads to incorrect reported attribute names.
 // Remap the attributes using the target device's manufacturer ID
 // if the header is lacking the information.
+// Note: Limit this feature to Legrand / 4129 only following regressions with Tuya devices
 function getCluster(frame: ZclFrame, deviceManufacturerID: number): Cluster {
     let cluster = frame.Cluster;
 
-    if (!frame?.Header?.manufacturerCode && frame?.Cluster && Number.isInteger(deviceManufacturerID)) {
+    if (!frame?.Header?.manufacturerCode && frame?.Cluster &&
+         Number.isInteger(deviceManufacturerID) && deviceManufacturerID == 4129) {
```

Comment:
I think there is no need to do the integer check here, just `deviceManufacturerID == 4129` should be good enough.

Reply:
The INT check is required due to some bogus tests, those may fail otherwise.
That check was already part of the first PR if I recall correctly.

Reply:
- How many tests are failing? Could those be fixed?
- I don't see a reason to revert #787 as that will break the Legrand use case?

Reply:
HI @Koenkk,
Sorry for the delay in getting back, i am currently on a business trip.

* There are 4 occurrences of missing manufacturerID at a quick glance. Those might be legitimate, I am not yet sure which tests are issuing those requests.
* I've removed the integer check, as you are right, it is definitly not required. I've gone through the history, and I really cannot recall why i have introduced that check really (quite sure it must have had a good reason!)..
So sorry for that, it is removed now!

* Yes, it would break the Legrand use case, if implementing the devices as specified by the Legrand documentation. Having that said, the documentation is inconsistent too.
As mentioned by @sjorge, the fix helps on "certain" devices, and may "break" others, depending on the firmware.
Would it make sense to add a new "meta" annotation, that could be enabled on a device basis in order to explicitly enable the remapping of the attributes instead of hardcoding the manufacturerID here ?